File: easy/moving-average-from-data-stream.cpp
class MovingAverage {
public:
    list<int> q;
    int size;
    int num = 0; 
    int sum = 0;
    MovingAverage(int s) {
        size = s;
    }
    
    double next(int val) {
        if (num < size){
            
            num ++;
            sum += val;
            q.push_back(val);
            return (double)sum / (double) num;
        } else {

            int first = q.front();
            q.pop_front();
            
            sum -= first; 
            sum += val;
            q.push_back(val);
            
            return (double) sum/ (double) num;
        };
    };
};

File: easy/moving-average-from-data-stream.py
from collections import deque

class MovingAverage:
    
    num = 0
    _sum = 0
    q = deque()
    

    def __init__(self, size: int):
        self.size = size
        
    def next(self, val: int) -> float:
        if self.num  < self.size:
            self.num += 1 
            self._sum += val
            self.q.append(val)
            return self._sum / self.num
        else:
            first = self.q.popleft()
            # print(first)
            self._sum -= first
            self._sum += val
            self.q.append(val)
            
            return self._sum / self.num
